src/main/java/htsjdk/samtools/util/GZIIndex.java

```diff
@@ -0,0 +1,365 @@
+
+    private final List<IndexEntry> entries;
+
+    // private constructors
+    private GZIIndex(final List<IndexEntry> entries) {
+        this.entries = entries;
+    }
+
+    /**
+     * Gets the number of blocks on the file.
+     *
+     * @return the number of blocks.
+     */
+    public int getNumberOfBlocks() {
+        return entries.size();
+    }
+
+    /**
+     * Gets an unmodifiable list with the index entries.
+     *
+     * @return index entries.
+     */
+    public List<IndexEntry> getIndexEntries() {
+        return Collections.unmodifiableList(entries);
+    }
+
+    /**
+     * Gets the virtual offset for seek with {@link BlockCompressedInputStream#seek(long)}.
+     *
+     * @param uncompressedOffset the file-offset.
+     *
+     * @return virtual offset for {@link BlockCompressedInputStream}.
+     * @see BlockCompressedFilePointerUtil
+     */
+    public long getVirtualOffsetForSeek(final long uncompressedOffset) {
+        if (uncompressedOffset == 0) {
+            return BlockCompressedFilePointerUtil.makeFilePointer(0, 0);
+        }
+
+        // list of entries at the uncompressed-offset (or afterwards)
+        final IndexEntry indexEntry = entries.stream()
+                // filter for only the ones where the uncompressed offset is larger than the requested
+                .filter(entry -> entry.getUncompressedOffset() >= uncompressedOffset)
+                // get the first, and if there is any, the first entry
+                // TODO - the first entry can be removed from the list if we use this code
+                // TODO - and we remove the method to get the index entries (unnecessary for most of the cases)
+                .findFirst().orElse(FIRST_MAPPING);
+
+        // convert to an offset within the block
+        final int blockOffset = Math.toIntExact(indexEntry.getUncompressedOffset() - uncompressedOffset);
+
+        return BlockCompressedFilePointerUtil.makeFilePointer(indexEntry.getCompressedOffset(), blockOffset);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == null || !(obj instanceof GZIIndex)) {
+            return false;
+        }
+        return this.entries.equals(((GZIIndex) obj).entries);
+    }
+
+    @Override
+    public int hashCode() {
+        return entries.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return "GZIIndex:" + StringUtil.join(", ", entries);
+    }
+
+    /**
+     * Writes this index into the requested path.
+     *
+     * @param output the output file.
+     *
+     * @throws IOException if an I/O error occurs.
+     */
+    public void writeIndex(final Path output) throws IOException {
+        if (output == null) {
+            throw new IllegalArgumentException("null output path");
+        }
+
+        // the first entry is never written - 0, 0
+        final int numberOfBlocksToWrite = getNumberOfBlocks() - 1;
```

Comment:
yeah, this complicated things, maybe we should just drop it since we probably don't need it for any thing

Reply:
I'll do in the next commit, to be sure that it does not break anything.